src/Connector.sol

```diff
@@ -78,6 +81,14 @@ contract CentrifugeConnector is Test {
         emit File(what, data);
     }
 
+    function file(bytes32 name, string memory domainName, uint32 domainId) public auth  {
+        if(name == "domain") {
+           domainLookup[keccak256(bytes(domainName))] = domainId;
+           emit File(name, domainName);
+        } else { revert ("unknown name");}
+        
+    }
+
     // --- Internal ---
     function addPool(uint64 poolId) public onlyRouter {
         Pool storage pool = pools[poolId];
```

Comment:
I can't leave a comment there, but I just realised line 108, setting the token price to 1.0 is a rather bad design. There is no guarantee that a token is added to another domain at this price.

I think there are two alternatives:
1. We can't initialize to `0` as a token can actually get a price of 0, so we initialize it to `type(uint256).max)`. Downside is that this will need to be accounted for in external integrations, so they don't accidentally use this for calculations and quickly overflow.
2. We pass the current token price when calling `addTranche`. @NunoAlexandre would be good to get your input here, but this seems viable to me, and would cleanly initiate the tranche on the domain with the right values.

Reply:
> We pass the current token price when calling addTranche. @NunoAlexandre would be good to get your input here, but this seems viable to me, and would cleanly initiate the tranche on the domain with the right values.


Sounds good to me yeah 👍

Reply:
Done ✅